lib/paper_trail/has_paper_trail.rb

```diff
@@ -87,26 +100,46 @@ def has_paper_trail(options = {})
             :order      => self.paper_trail_version_class.timestamp_sort_order
         end
 
-        options[:on] ||= [:create, :update, :destroy]
-
-        # Wrap the :on option in an array if necessary. This allows a single
-        # symbol to be passed in.
-        options_on = Array(options[:on])
-
-        after_create  :record_create, :if => :save_version? if options_on.include?(:create)
-        if options_on.include?(:update)
-          before_save   :reset_timestamp_attrs_for_update_if_needed!, :on => :update
-          after_update  :record_update, :if => :save_version?
-          after_update  :clear_version_instance!
-        end
-        after_destroy :record_destroy, :if => :save_version? if options_on.include?(:destroy)
-
         # Reset the transaction id when the transaction is closed.
         after_commit :reset_transaction_id
         after_rollback :reset_transaction_id
         after_rollback :clear_rolled_back_versions
       end
 
+      def setup_callbacks_from_options(options_on = [])
+        options_on.each do |option|
+          send "paper_trail_on_#{option}"
+        end
+
+        paper_trail_options[:on] = options_on
```

Comment:
Isn't this line unnecessary because of the assignment in `setup_model_for_paper_trail`? (This assignment: `self.paper_trail_options = options.dup`)

Also, this line has got me thinking .. what about the case like this:

```
has_paper_trail on: []
paper_trail_on_destroy
```

The result should be that `paper_trail_options[:on] == [:destroy], but I don't think this PR will do that.  Shouldn't the callback methods like `paper_trail_on_destroy` append to `paper_trail_options[:on]`?